src/Storages/StorageMergeTree.cpp

```diff
@@ -1086,6 +1080,14 @@ bool StorageMergeTree::scheduleDataProcessingJob(IBackgroundJobExecutor & execut
         }, PoolType::MERGE_MUTATE});
         return true;
     }
+    if (has_mutations)
```

Comment:
Why this has been changed from the original code?

```suggestion
    if (!mutate_entry && has_mutations)
```

If `mutate_entry` is set, then the notification will happen from other place, so you don't need to notify here.
Am I missing something?

(p.s. this also moves the code into another place and increases the diff)

Reply:
If `merge_entry` is set, `mutate_entry` can never be true, but you fire event in vain in this case. So I simply changed `!mutate_entry && has_mutations` -> `!merge_entry && !mutate_entry && has_mutations`, effectively.

Diff, yes, but I think this is a proper order of processing.

Reply:
`mutate_entry` is not true at this point, because of `return` above.

Reply:
>mutate_entry is not true at this point, because of return above. And merge_entry as well.

Oh, I see now, thanks!